server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -878,36 +890,33 @@ private static Mapper getLeafMapper(final DocumentParserContext context,
         // if a leaf field is not mapped, and is defined as a runtime field, then we
         // don't create a dynamic mapping for it and don't index it.
         String fieldPath = context.path().pathAsText(fieldName);
-        MappedFieldType fieldType = context.mappingLookup().getFieldType(fieldPath);
-        if (fieldType != null) {
-            RuntimeField runtimeField = context.root().getRuntimeField(fieldPath);
-            if (runtimeField != null) {
-                assert fieldType.hasDocValues() == false && fieldType.isAggregatable() && fieldType.isSearchable();
```

Comment:
I don't know if its useful to keep this assertion, no strong opinion, just curious if you deleted it for some specific reason.

Reply:
I think it was an extra check that this was definitely a runtime field.  I don't think it's necessary any longer because we now know where the shadowing logic is pulling fields from.